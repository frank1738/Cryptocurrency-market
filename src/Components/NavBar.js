import React from 'react';
import { FaCoins } from 'react-icons/fa';
import { NavLink } from 'react-router-dom';

const NavBar = ({ page, set }) => {
  return (
    <nav>
      <div className="nav-bar">
        <NavLink to="/">
          {page === 'home' ? (
            <i className="uil uil-bitcoin coin-icon"></i>
          ) : (
            <i
              className="uil uil-angle-left arrow"
              onClick={() => set('home')}
            ></i>
          )}
        </NavLink>
        <div className="logo">
          <FaCoins className="icon" />
          <h1 className="nav-header">
            Coin<span className="purple">Prices</span>
          </h1>
        </div>
        <div className="mic">
          <i className="uil uil-microphone micro nav-icon"></i>
          <i className="uil uil-cog nav-icon"></i>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
