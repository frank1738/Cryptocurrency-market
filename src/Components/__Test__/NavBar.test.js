import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import NavBar from '../NavBar';
import { store } from '../../Redux/configureStore';
import { BrowserRouter } from 'react-router-dom';

describe('NavBar test', () => {
  test('should render NavBar properly', () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
