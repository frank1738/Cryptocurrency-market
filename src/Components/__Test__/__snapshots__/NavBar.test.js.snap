// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar test should render NavBar properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <div
          class="nav-bar"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <i
              class="uil uil-angle-left arrow"
            />
          </a>
          <div
            class="logo"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 405.3V448c0 35.3 86 64 192 64s192-28.7 192-64v-42.7C342.7 434.4 267.2 448 192 448S41.3 434.4 0 405.3zM320 128c106 0 192-28.7 192-64S426 0 320 0 128 28.7 128 64s86 64 192 64zM0 300.4V352c0 35.3 86 64 192 64s192-28.7 192-64v-51.6c-41.3 34-116.9 51.6-192 51.6S41.3 334.4 0 300.4zm416 11c57.3-11.1 96-31.7 96-55.4v-42.7c-23.2 16.4-57.3 27.6-96 34.5v63.6zM192 160C86 160 0 195.8 0 240s86 80 192 80 192-35.8 192-80-86-80-192-80zm219.3 56.3c60-10.8 100.7-32 100.7-56.3v-42.7c-35.5 25.1-96.5 38.6-160.7 41.8 29.5 14.3 51.2 33.5 60 57.2z"
              />
            </svg>
            <h1
              class="nav-header"
            >
              Coin
              <span
                class="purple"
              >
                Prices
              </span>
            </h1>
          </div>
          <div
            class="mic"
          >
            <i
              class="uil uil-microphone micro nav-icon"
            />
            <i
              class="uil uil-cog nav-icon"
            />
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <div
        class="nav-bar"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <i
            class="uil uil-angle-left arrow"
          />
        </a>
        <div
          class="logo"
        >
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 405.3V448c0 35.3 86 64 192 64s192-28.7 192-64v-42.7C342.7 434.4 267.2 448 192 448S41.3 434.4 0 405.3zM320 128c106 0 192-28.7 192-64S426 0 320 0 128 28.7 128 64s86 64 192 64zM0 300.4V352c0 35.3 86 64 192 64s192-28.7 192-64v-51.6c-41.3 34-116.9 51.6-192 51.6S41.3 334.4 0 300.4zm416 11c57.3-11.1 96-31.7 96-55.4v-42.7c-23.2 16.4-57.3 27.6-96 34.5v63.6zM192 160C86 160 0 195.8 0 240s86 80 192 80 192-35.8 192-80-86-80-192-80zm219.3 56.3c60-10.8 100.7-32 100.7-56.3v-42.7c-35.5 25.1-96.5 38.6-160.7 41.8 29.5 14.3 51.2 33.5 60 57.2z"
            />
          </svg>
          <h1
            class="nav-header"
          >
            Coin
            <span
              class="purple"
            >
              Prices
            </span>
          </h1>
        </div>
        <div
          class="mic"
        >
          <i
            class="uil uil-microphone micro nav-icon"
          />
          <i
            class="uil uil-cog nav-icon"
          />
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
